pyrefly/lib/alt/function.rs

```diff
@@ -110,7 +110,29 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
             }
         } else {
             let mut acc = Vec::new();
+            let implementation_range = def.id_range;
             let mut first = def;
+            let mut has_overload_successor = false;
+            
+            let binding = self.bindings().get(idx);
+            let mut current_successor = binding.successor;
+            while let Some(succ_key_idx) = current_successor {
+                let succ_def = self.get_idx(succ_key_idx);
+                if succ_def.metadata.flags.is_overload {
+                    has_overload_successor = true;
+                    break;
+                }
+                current_successor = self.bindings().get(succ_key_idx).successor;
+            }
+            if has_overload_successor && !skip_implementation {
+                self.error(
+                    errors,
+                    implementation_range,
+                    ErrorKind::InvalidOverload,
+                    None,
+                    "Function implementation must come after all @overload declarations. ".to_owned(),
```

Comment:
I like this error message, but I think it's confusing that this error and "Overloaded function must have an implementation" are both reported; we should report one or the other.

What I'd suggest doing is: when we encounter an `@overload`-decorated signature with no successors, we loop through the predecessors to see if there's an implementation among them. If there is, then we report this error rather than the "Overloaded function must have an implementation" error. Here's an example of how to loop through predecessors: https://github.com/facebook/pyrefly/blob/4a08127ee842d006904558dca4baecf9a3b12251/pyrefly/lib/alt/function.rs#L485-L502

Reply:
Sorry for the delay! I somehow missed a bunch of GitHub notifications this past week. I'll review this tomorrow.